src/tribol/mesh/MfemData.hpp

```diff
@@ -0,0 +1,1390 @@
+  );
+
+  /**
+   * @brief First mesh identifier
+   */
+  integer mesh_id_1_;
+
+  /**
+   * @brief Second mesh identifier
+   */
+  integer mesh_id_2_;
+
+  /**
+   * @brief Volume mesh of parent domain
+   */
+  mfem::ParMesh& mesh_;
+
+  /**
+   * @brief Mesh boundary attributes identifying first mesh
+   */
+  std::set<integer> attributes_1_;
+
+  /**
+   * @brief Mesh boundary attributes identifying second mesh
+   */
+  std::set<integer> attributes_2_;
+
+  /**
+   * @brief Submesh containing boundary elements of both contact surfaces
+   */
+  mfem::ParSubMesh submesh_;
+
+  /**
+   * @brief Coordinates grid function and transfer operators
+   */
+  ParentField coords_;
+
+  /**
+   * @brief (Coarse) submesh grid function to temporarily hold values being
+   * transferred
+   */
+  mfem::ParGridFunction submesh_xfer_gridfn_;
+
+  /**
+   * @brief Refinement factor for refined mesh
+   */
+  integer lor_factor_;
+
+  /**
+   * @brief Contains refined submesh if low-order refinement is being used;
+   * nullptr otherwise
+   */
+  std::unique_ptr<mfem::ParMesh> lor_submesh_;
+
+  /**
+   * @brief Contains low-order refined submesh transfer operators if low-order
+   * refinement is being used; nullptr otherwise
+   */
+  std::unique_ptr<SubmeshLORTransfer> lor_xfer_;
+
+  /**
+   * @brief Type of elements on the contact mesh
+   */
+  InterfaceElementType elem_type_;
+
+  /**
+   * @brief Number of vertices on each element in the contact mesh
+   */
+  integer num_verts_per_elem_;
+
+  /**
+   * @brief Contains velocity grid function and transfer operators if set;
+   * nullptr otherwise
+   */
+  std::unique_ptr<ParentField> velocity_;
+
+  /**
+   * @brief UpdateData object created upon call to UpdateMeshData()
+   */
+  std::unique_ptr<UpdateData> update_data_;
+
+  /**
+   * @brief Nodal response grid function on the redecomp mesh
+   */
+  mfem::GridFunction redecomp_response_;
+
+  /**
+   * @brief Merges two STL containers
+   * 
+   * @tparam T container type
+   * @param container_1 First container
+   * @param container_2 Second container
+   * @return T merged container
+   */
+  template <typename T>
+  static T mergeContainers(T container_1, T container_2)
+  {
+    auto merged = container_1;
+    merged.insert(container_2.begin(), container_2.end());
+    return merged;
+  }
+
+  /**
+   * @brief Converts a std::set to an mfem::Array
+   * 
+   * @tparam T type held in the set and array
+   * @param orig original set
+   * @return mfem::Array<T> output array holding entries in orig
+   */
+  template <typename T>
+  static mfem::Array<T> arrayFromSet(std::set<T> orig)
+  {
+    auto array = mfem::Array<T>();
+    array.Reserve(static_cast<int>(orig.size()));
+    for (const auto& val : orig)
+    {
+      array.Append(val);
+    }
+    return array;
+  }
+};
+
+/**
+ * @brief Stores MFEM and transfer data associated with submesh pressure and gap
+ * fields 
+ */
+class MfemSubmeshData
```

Comment:
Similar to MortarData etc., it looks like this treats mortar specific data, not MfemSubmeshData generally for any future method. Should we call this MfemSubmeshMortarData? Common plane would not use this, and I separated out method specific data to prevent confusion and data overload.

Reply:
It's specific to Lagrange multiplier fields, so I'd say keep it as is.  Future non-mortar contact methods might use it.